crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,7 +287,125 @@ impl Executor {
                     }
                 }
             }
+        };
+
+        match &reason {
+            Ok(reason) => {
+                info!(reason, "shutting down");
+            }
+            Err(reason) => {
+                error!(%reason, "shutting down");
+                // we error out mainly because of a failure to submit a bundle to the sequencer
+                // we do not want to proceed with the shutdown process in this case
+                return Err(eyre!(reason.to_string()));
```

Comment:
@SuperFluffy what do you think of the way i m returning the error reason?

Reply:
Also note for reviewer: I am choosing to exit at this point rather than continuing, because its very unlikely we even reach this point because we retry almost infinitely. 
In a practical scenario, we would observe the retries and manually shutdown composer.

Reply:
The return at this point (including skipping the drain) is fine - you know if it makes sense to attempt to drain or not.

But I don't see a point formatting the error at this point?. Just `return Err(reason)`